src/transforms/transform-code-scanning-alert.ts

```diff
@@ -0,0 +1,93 @@
+import { jiraIssueKeyParser } from "utils/jira-utils";
+import {Context} from "probot/lib/context";
+import {JiraRemoteLinkData, JiraRemoteLinkStatusAppearance} from "interfaces/jira";
+
+const MAX_STRING_LENGTH = 255;
+
+const getPullRequestTitle = async (repoName: string, prId: number, repoOwner: string, context: Context): Promise<string> => {
+	const response = await context.github.pulls.get({
+		owner: repoOwner,
+		repo: repoName,
+		pull_number: prId
+	});
+
+	if (response.status !== 200) {
+		context.log.warn(`Received error when querying for Pull Request information: ${JSON.stringify(response.data)}`);
+		return "";
+	} else {
+		return response.data.title;
+	}
+}
+
+const getEntityTitle = async (ref: string, repoName: string, repoOwner: string, context: Context): Promise<string> => {
+	// ref can either be a branch reference or a PR reference
+	const components = ref.split("/")
+	switch (components[1]) {
+		case "heads": // branch
+			// The branch name may contain forward slashes! Rejoin them
+			return Promise.resolve(components.slice(2).join("/"));
+		case "pull": // pull request
+			return await getPullRequestTitle(repoName, parseInt(components[2]), repoOwner, context);
+		default:
+			context.log.error(`Could not interpret reference from code_scanning_alert: ${ref}`);
+			return Promise.resolve("");
```

Comment:
just `return "";` will do, however, are you sure you want to continue if you can't find the entity name?

Reply:
Yup that's fine because if we don't find an entity type we just won't find any issue keys and will return the function on line 70.